systemd/systemd.go

```diff
@@ -154,6 +163,30 @@ func (*systemd) DaemonReload() error {
 	return err
 }
 
+func (s *systemd) DaemonReloadIfNeeded(serviceNames ...string) error {
+	needReload := false
+	for _, service := range serviceNames {
+		// A status error will happen if the service does not exist anymore.
+		// We do nothing in that case, as in some cases that is the expected
+		// state of the service when this function is called.
```

Comment:
I realize this is still WIP, but as a heads up the fact we are doing that sounds like a problem on itself, if that's indeed the case. We shouldn't try to reload a daemon that doesn't exist, and we should hopefully have metadata to tell whether that's the case at the call site.